File: ECommerceApi/Models/Order.cs
namespace ECommerceApi.Models
{
    public enum OrderStatus
    {
        Pending,
        Processing,
        Shipped,
        Delivered,
        Cancelled,
        Refunded
    }

    public enum PaymentStatus
    {
        Pending,
        Paid,
        Failed,
        Refunded
    }

    public enum PaymentMethod
    {
        CreditCard,
        PayPal,
        Stripe,
        BankTransfer,
        CashOnDelivery
    }

    public class Order
    {
        public int Id { get; set; }

        [Required]
        public string OrderNumber { get; set; } = string.Empty;

        [Required]
        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Pending;
        public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.CreditCard;

        public string? PaymentIntenId { get; set; }

        [Required]
        public decimal TotalAmount { get; set; }
        public decimal TaxAmount { get; set; } = 0;
        public decimal ShippingCost { get; set; } = 0;
        public decimal DiscountAmount { get; set; } = 0;
        public decimal FinalAmount => TotalAmount + TaxAmount + ShippingCost - DiscountAmount;

        // Informations de livraison
        public string ShippingAddress { get; set; } = string.Empty;
        public string ShippingCity { get; set; } = string.Empty;
        public string ShippingPostalCode { get; set; } = string.Empty;
        public string ShippingCountry { get; set; } = string.Empty;
        public string? TrackingNumber { get; set; }

        // Informations de facturation
        public string BillingAddress { get; set; } = string.Empty;
        public string BillingCity { get; set; } = string.Empty;
        public string BillingPostalCode { get; set; } = string.Empty;
        public string BillingCountry { get; set; } = string.Empty;

        public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
        public DateTime? PaidAt { get; set; }
        public DateTime? ShippedAt { get; set; }
        public DateTime? DeliveredAt { get; set; }
    }
    
    public class OrderItem
    {
        public int Id { get; set; }
        
        [Required]
        public int OrderId { get; set; }
        public Order Order { get; set; } = null!;
        
        [Required]
        public int ProductId { get; set; }
        public Product Product { get; set; } = null!;
        
        [Required]
        public int Quantity { get; set; }
        
        [Required]
        public decimal UnitPrice { get; set; } // Prix au moment de l'achat
        
        public decimal TotalPrice => Quantity * UnitPrice;
        
        // Pour les reviews
        public bool IsReviewed { get; set; } = false;
    }

}

File: ecommerce-frontend/types/order.ts
import { User } from './user';
import { Product } from './product';

// ✅ Enum avec statuts numériques ET string pour compatibilité
export enum OrderStatus {
  Pending = 'Pending',
  Processing = 'Processing',
  Shipped = 'Shipped',
  Delivered = 'Delivered',
  Cancelled = 'Cancelled',
  Refunded = 'Refunded',
  ReturnRequested = 'ReturnRequested',
}

export enum PaymentStatus {
  Pending = 'Pending',
  Paid = 'Paid',
  Failed = 'Failed',
  Refunded = 'Refunded'
}

export enum PaymentMethod {
  Card = 'Card',
  PayPal = 'PayPal',
  BankTransfer = 'BankTransfer'
}

// ✅ Mapping statuts numériques <-> string
export const OrderStatusMap = {
  0: OrderStatus.Pending,
  1: OrderStatus.Processing,
  2: OrderStatus.Shipped,
  3: OrderStatus.Delivered,
  4: OrderStatus.Cancelled,
  5: OrderStatus.Refunded,
  6: OrderStatus.ReturnRequested,
} as const;

export const OrderStatusReverseMap: Record<OrderStatus, number> = {
  [OrderStatus.Pending]: 0,
  [OrderStatus.Processing]: 1,
  [OrderStatus.Shipped]: 2,
  [OrderStatus.Delivered]: 3,
  [OrderStatus.Cancelled]: 4,
  [OrderStatus.Refunded]: 5,
  [OrderStatus.ReturnRequested]: 6,
};

export interface OrderItem {
  id: number;
  orderId: number;
  productId: number;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  isReviewed: boolean;
  product?: Product;
}

export interface Order {
  id: number;
  orderNumber: string;
  userId: number;
  user?: User;
  status: OrderStatus;
  paymentStatus: PaymentStatus;
  paymentMethod: PaymentMethod;
  paymentIntentId?: string;
  
  // Montants
  subtotal: number;
  taxAmount: number;
  shippingCost: number;
  discountAmount: number;
  totalAmount: number;
  finalAmount: number;
  
  // Adresses
  shippingAddress: string;
  shippingCity: string;
  shippingPostalCode: string;
  shippingCountry: string;
  billingAddress: string;
  billingCity: string;
  billingPostalCode: string;
  billingCountry: string;
  
  // Dates
  createdAt: string;
  updatedAt?: string;
  paidAt?: string;
  shippedAt?: string;
  deliveredAt?: string;
  
  // Tracking
  trackingNumber?: string;
  
  // Relations
  items: OrderItem[];
}

export interface CreateOrderDto {
  paymentMethod: number;
  shippingAddress: string;
  shippingCity: string;
  shippingPostalCode: string;
  shippingCountry: string;
  billingAddress?: string;
  billingCity?: string;
  billingPostalCode?: string;
  billingCountry?: string;
  notes?: string;
  taxAmount?: number;
  shippingCost?: number;
  discountAmount?: number;
}

// ⭐ NOUVEAU — DTO pour le breakdown de livraison par boutique
export interface OrderShopShippingDto {
  shopId: number;
  shopName?: string;
  shippingMethodName: string;
  shippingCost: number;
  subtotal: number;
  minDays: number;   // ⭐ AJOUT
  maxDays: number;   // ⭐ AJOUT
}

// ⭐ MODIFICATION — Ajout de shippingBreakdown
export interface OrderResponseDto {
  id: number;
  orderNumber: string;
  userId: number;
  username: string;
  userEmail?: string;
  status: OrderStatus | string | number;
  paymentStatus: PaymentStatus;
  paymentMethod: PaymentMethod;
  paymentIntentId?: string;
  totalAmount: number;
  taxAmount: number;
  shippingCost: number;
  discountAmount: number;
  finalAmount: number;
  shippingAddress: string;
  shippingCity: string;
  shippingPostalCode: string;
  shippingCountry: string;
  billingAddress: string;
  billingCity: string;
  billingPostalCode: string;
  billingCountry: string;
  trackingNumber?: string;
  createdAt: string;
  paidAt?: string;
  shippedAt?: string;
  deliveredAt?: string;
  items: OrderItemDto[];
  // ⭐ NOUVEAU — Détail des frais de livraison par boutique
  shippingBreakdown: OrderShopShippingDto[];
}

export interface OrderItemDto {
  id: number;
  productId: number;
  productName: string;
  productImage?: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  shopId?: number;
  shopName?: string;
  isReviewed: boolean;
}

export interface UpdateOrderStatusDto {
  status: OrderStatus | number | string;
}

export interface CreatePaymentIntentDto {
  orderId: number;
}

export interface ConfirmPaymentDto {
  orderId: number;
  paymentIntentId: string;
}

export interface RefundRequestDto {
  amount?: number;
  reason?: string;
}

export interface PaymentIntentResponseDto {
  id: string;
  clientSecret: string;
  amount: number;
  currency: string;
  status: string;
}

export interface PaymentIntent {
  id: string;
  amount: number;
  currency: string;
  status: string;
  clientSecret: string;
  metadata?: Record<string, string>;
}